src/Host/Protocol/Impl/RUserProfile/RUserProfileServicesImpl.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Security.Principal;
+using System.Text;
+using System.Runtime.InteropServices;
+using Microsoft.Common.Core.OS;
+using Microsoft.Common.Core.Security;
+using Microsoft.Extensions.Logging;
+using static Microsoft.R.Host.Protocol.NativeMethods;
+
+namespace Microsoft.R.Host.Protocol {
+    class RUserProfileServicesImpl : IUserProfileServices {
```

Comment:
Or actually that is also wrong, since the request loop is not there. It would be something like `RUserProfileRequestHandler`.